### PR TITLE
chore: use latest image to deploy windows e2e vmss

### PR DESCRIPTION
**Pull request by junjiezhang1997**
**What type of PR is this?**

**What this PR does / why we need it**:
Use latest image to deploy windows e2e vmss.

**Which issue(s) this PR fixes**:

Fixes #

**Requirements**:
- [x] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Special notes for your reviewer**:

**Release note**:
```
none
```

### CODE REVIEW

File: e2e/windows/e2e-scenario.sh [hunk 1]
```diff
@@ -140,9 +140,7 @@ MC_WIN_VMSS_NAME=$(az vmss list -g $MC_RESOURCE_GROUP_NAME --query "[?contains(n
 VMSS_RESOURCE_Id=$(az resource show --resource-group $MC_RESOURCE_GROUP_NAME --name $MC_WIN_VMSS_NAME --resource-type Microsoft.Compute/virtualMachineScaleSets --query id --output tsv)
 
 az group export --resource-group $MC_RESOURCE_GROUP_NAME --resource-ids $VMSS_RESOURCE_Id --include-parameter-default-value > test.json
-# TODO: use prod image
-# IMAGE_REFERENCE=$(jq -c '.resources[0].properties.virtualMachineProfile.storageProfile.imageReference.id' test.json)
-IMAGE_REFERENCE="[concat(parameters('galleries_AKSWindows_externalid'), '/images/windows-e2e-test-$WINDOWS_E2E_IMAGE/versions/2023.02.07')]"
+IMAGE_REFERENCE=$(az sig image-version list --gallery-image-definition "windows-e2e-test-$WINDOWS_E2E_IMAGE" --gallery-name $IMAGE_GALLERY_NAME --resource-group $IMAGE_RESOURCE_GROUP --subscription $IMAGE_SUBSCRIPTION_ID  | jq -r '.[-1].id')
```
[the remark this answers is not in the document]
Comment (junjiezhang1997) [reply e2e/windows/e2e-scenario.sh:143]: Yes. The last result is the latest image version. I'm trying to find any doc that clarifies the list order or any command to sort the list result.

[the remark this answers is not in the document]
Comment (AbelHu) [reply e2e/windows/e2e-scenario.sh:143]: Could we use `az sig image-definition` to get the resource id of image definition and then append `/versions/latest`?

[the remark this answers is not in the document]
Comment (junjiezhang1997) [reply e2e/windows/e2e-scenario.sh:143]: Do you mean use image reference like "/subscriptions/$SUBSCRIPTION/resourceGroups/$RESOURCE_GROUP/providers/Microsoft.Compute/galleries/$GALLERY_NAME/images/windows-2022-containerd/versions/latest"?

Not sure whether such image reference would work. I can test it manually.

File: e2e/windows/e2e-scenario.sh [hunk 2]
```diff
@@ -140,10 +140,8 @@ MC_WIN_VMSS_NAME=$(az vmss list -g $MC_RESOURCE_GROUP_NAME --query "[?contains(n
 VMSS_RESOURCE_Id=$(az resource show --resource-group $MC_RESOURCE_GROUP_NAME --name $MC_WIN_VMSS_NAME --resource-type Microsoft.Compute/virtualMachineScaleSets --query id --output tsv)
 
 az group export --resource-group $MC_RESOURCE_GROUP_NAME --resource-ids $VMSS_RESOURCE_Id --include-parameter-default-value > test.json
-# TODO: use prod image
-# IMAGE_REFERENCE=$(jq -c '.resources[0].properties.virtualMachineProfile.storageProfile.imageReference.id' test.json)
-IMAGE_REFERENCE="[concat(parameters('galleries_AKSWindows_externalid'), '/images/windows-e2e-test-$WINDOWS_E2E_IMAGE/versions/2023.02.07')]"
-IMAGE_EXTERNALID="/subscriptions/$IMAGE_SUBSCRIPTION_ID/resourceGroups/akswinvhdbuilderrg/providers/Microsoft.Compute/galleries/AKSWindows"
+IMAGE_REFERENCE="[concat(parameters('galleries_AKSWindows_externalid'), '/images/windows-e2e-test-$WINDOWS_E2E_IMAGE/versions/latest')]"
```
Comment (AbelHu) [inline e2e/windows/e2e-scenario.sh:143]: Suggest to use IMAGE_REFERENCE and remove IMAGE_EXTERNALID to simplify the code

Comment (junjiezhang1997) [reply e2e/windows/e2e-scenario.sh]: Thanks for the advice. Done.

File: e2e/windows/e2e-scenario.sh [hunk 3]
```diff
@@ -159,13 +156,12 @@ jq --argjson JsonForVnet "$WINDOWS_VNET" \
...
     --arg ValueForImageReference "$IMAGE_REFERENCE" \
     --arg ValueForAdminPassword "$WINDOWS_PASSWORD" \
     --arg ValueForCustomData "$CUSTOM_DATA" \
     --arg ValueForCSECmd "$CSE_CMD" \
     --arg ValueForVMSS "$DEPLOYMENT_VMSS_NAME" \
-    '.parameters += $JsonForVnet | .parameters += $JsonForLB | .parameters.galleries_AKSWindows_externalid.defaultValue=$ValueForImageExternalID | .resources[0] += $JsonForIdentity | .resources[0] += $JsonForSKU | .resources[0].properties.virtualMachineProfile.storageProfile+=$JsonForOSDisk | .resources[0].properties.virtualMachineProfile.networkProfile.networkInterfaceConfigurations[0] += $JsonForNetwork | .resources[0].properties.virtualMachineProfile.storageProfile.imageReference.id=$ValueForImageReference | .resources[0].properties.virtualMachineProfile.osProfile.adminPassword=$ValueForAdminPassword | .resources[0].properties.virtualMachineProfile.osProfile.customData=$ValueForCustomData | .resources[0].properties.virtualMachineProfile.extensionProfile.extensions[0].properties.settings.commandToExecute=$ValueForCSECmd | .parameters.virtualMachineScaleSets_akswin30_name.defaultValue=$ValueForVMSS' \
```
Comment (AbelHu) [inline e2e/windows/e2e-scenario.sh:168]: could we also delete galleries_AKSWindows_externalid in [e2e/windows/windows_vmss_template.json](https://github.com/Azure/AgentBaker/blob/cb86e406100687b82f50ac03d2d248ef21861c64/e2e/windows/windows_vmss_template.json#L9)?

Comment (junjiezhang1997) [reply e2e/windows/e2e-scenario.sh:168]: Missed this part before. Thanks for the reminder.

**Pull request merged**